packages/node/src/indexer/indexer.manager.ts

```diff
@@ -352,12 +364,12 @@ export class IndexerManager {
         const processor = asSecondLayerHandlerProcessor_1_0_0(
           plugin.handlerProcessors[handler.kind],
         );
-
         try {
           return processor.filterProcessor({
             filter: handler.filter,
             input: data,
             ds,
+            assets,
```

Comment:
I'm guessing we need to decode the input in order to filter. Is that going to slow things down as it will be decoded twice now (filter and transform)? 

Ideally we don't need to decode with the filter bit if we do can we somehow share the decoded data? Maybe keeping it in memory in the processor